zipinfo.js

```diff
@@ -87,11 +109,56 @@ ZipInfo.getEntries = function(data, dataStartOffset) {
     var utfLabel = (bitFlag & 0x800) ? 'utf-8' : 'ascii';
     filename = ZipInfo._decodeFilename(filename, utfLabel);
 
+    var localEntryOffset = view.getUint32(offset + 42, true);
+
+    // ZIP64 support
+    if (compressedSize === 0xFFFFFFFF ||
+        uncompressedSize === 0xFFFFFFFF ||
+        localEntryOffset === 0xFFFFFFFF) {
+
+      var extraFieldOffset = offset + 46 + fileNameLength;
+      var efEnd = extraFieldOffset + extraFieldLength - 3;
+
+      while (extraFieldOffset < efEnd) {
```

Comment:
This could get stuck in an infinite loop, because of the use of 64 bit numbers, which goes beyond the precision of a safe JavaScript integer.

When you get in that territory, mathematically incorrect results could happen:

```
a = Math.pow(2, 60);
a === a - 1  // true, while it should be false.
```

Reply:
Ah good point! Probably safer to just loop at most 3 times to read those fields

Reply:
That would only work if the zip file packs all extra fields at the front. While the loop will terminate, the result may not match your expectations.